xls/contrib/mlir/IR/xls_ops.td

```diff
@@ -135,6 +135,36 @@ def Xls_TranslationLinkage : Xls_Attr<"TranslationLinkage"> {
   let attrName = "xls.translation_linkage";
 }
 
+def Xls_FlopKindNone        : I32EnumAttrCase<"kNone", 0, "none">;
+def Xls_FlopKindFlop        : I32EnumAttrCase<"kFlop", 1, "flop">;
+def Xls_FlopKindSkid        : I32EnumAttrCase<"kSkid", 2, "skid">;
+def Xls_FlopKindZeroLatency : I32EnumAttrCase<"kZeroLatency", 3, "zero-latency">;
+def Xls_FlopKind : I32EnumAttr<"FlopKind", "XLS flop kind",
+  [Xls_FlopKindNone, Xls_FlopKindFlop, Xls_FlopKindSkid, Xls_FlopKindZeroLatency]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::xls";
```

Comment:
Could we perhaps document the cases here? It goes a bit above what you;ve already done, but these  in isolation one has to go look up again :) (perhaps link to [https://google.github.io/xls/codegen_options/](https://google.github.io/xls/codegen_options/) and then copy the first lines over here so easy to see/I need to enable the documentation generation for the dialect still somewhere, but would be useful then))